src/storage/src/hummock/iterator/forward_user.rs

```diff
@@ -209,16 +165,14 @@ impl<I: HummockIterator<Direction = Forward>> UserIterator<I> {
             }
         };
 
-        // Handle multi-version
-        self.last_key = FullKey::default();
-        // Handles range scan when key > end_key
-        self.next().await
+        self.try_advance_to_next_valid().await
     }
 
     /// Resets the iterating position to the first position where the key >= provided key.
     pub async fn seek(&mut self, user_key: UserKey<&[u8]>) -> HummockResult<()> {
         // Reset
-        self.out_of_range = false;
+        self.is_current_pos_valid = false;
```

Comment:
+1

Reply:
> Seems unnecessary to set the is_current_pos_valid

I am okay both ways but I just think It is safer to not assume `try_advance_to_next_valid` will manipulate `is_current_pos_valid` and reset the states on `rewind` and `seek`.

> check self.iterator.is_valid() and self.key_out_of_range(), because in try_advance_to_next_valid can do the same thing anyway. Same for rewind.

I think the reason why we need to check them before `try_advance_to_next_valid` is to avoid the `delete_range_iter.seek/rewind`, which can be expensive, if the iterator is not valid. I will leave a note so that we can adopt your suggestion after we completely deprecate delete range. It took me a while to realize the current trick.

Reply:
> Seems unnecessary to set the is_current_pos_valid

Just realize that we must reset the flag on `seek` and `rewind` because we need to make sure `is_valid` returns false if there is a error happening in inner iterator seek/rewind.